flatpak-package.patch

```diff
@@ -0,0 +1,202 @@
+diff --git a/packages/Makefile b/packages/Makefile
+index e9964496c..b877e20e0 100644
+--- a/packages/Makefile
++++ b/packages/Makefile
+@@ -10,7 +10,7 @@
+ CWD := $(shell pwd)
+ 
+ cvs_root_dir := $(shell cd $(CWD)/.. && pwd)
+-DESTDIR = $(CWD)/build
++DESTDIR = /
+ BUILDLAYOUT_DIR = $(CWD)
+ 
+ 
+diff --git a/packages/linux_make/Makefile b/packages/linux_make/Makefile
+index 91fea1a84..04ca0109b 100755
+--- a/packages/linux_make/Makefile
++++ b/packages/linux_make/Makefile
+@@ -5,28 +5,14 @@
+ CWD := $(shell pwd)
+ 
+ # DESTDIR needs the trailing slash
+-DESTDIR = $(CWD)/build/
++DESTDIR = /
+ cvs_root_dir := $(shell cd $(CWD)/../.. && pwd)
+ BUILDLAYOUT_DIR = $(cvs_root_dir)/packages
+ 
+ include $(BUILDLAYOUT_DIR)/Makefile.buildlayout
+ 
+-# if machine has dpkg-deb, build a Debian package
+-ifeq ($(shell test -x /usr/bin/dpkg-shlibdeps && echo YES),YES)
+-  PACKAGE_TYPE = deb
+-else
+-  PACKAGE_TYPE = tarbz2
+-endif
+-
+-# for command-line UNIX versions, you need to install Pd into the same
+-# directory as $(prefix) otherwise Pd won't be able to find extra, doc, etc.
+-ifeq ($(PACKAGE_TYPE),deb)
+-  prefix = /usr
+-  # generates the dependencies for all externals and pd itself
+-  PACKAGE_DEPENDS = $(shell find $(DESTDIR) \( -name "*.pd_linux" -or -name pd-gui -or -wholename \*bin/pd \) -print0 | xargs -0 dpkg-shlibdeps -O | sed 's|shlibs:Depends=|, |' )
+-else
+-  prefix = /usr/local
+-endif
++PACKAGE_TYPE = flatpak
++prefix = /app
+ 
+ # to control whether the debug symbols are kept
+ #STRIP = strip --strip-unneeded -R .note -R .comment
+@@ -112,59 +98,44 @@ tarbz2: installer_makefile installer_readme
+ 	mv $(DESTDIR)$(PACKAGE_NAME) $(DESTDIR)$(prefix)
+ 
+ 
+-deb: DEB_BUILD_ARCH := $(shell dpkg-architecture -qDEB_BUILD_ARCH)
+-deb: DEB_PD_VERSION := $(shell echo $(PD_VERSION) | sed 's|\(.*\)-extended-\(.*\)|\1~cvs\2-1|')
+-deb: $(bindir)
++flatpak: $(bindir)
+ # free desktop menu, mime, and icon support
+-	install -d $(DESTDIR)/usr/share/icons/hicolor/128x128/apps
++	install -d $(DESTDIR)/app/share/icons/hicolor/128x128/apps
```

Comment:
ideally an upstream patch here would honour `$PREFIX` like it should.

Reply:
yeah, the upstream buildsystem needs a lot of work.  The goal is to sync more with net.purrdata.Purrdata, since there is a lot of shared pieces.  That will look different than patching those particular files.